pyiceberg/table/__init__.py

```diff
@@ -209,6 +221,48 @@ def set_properties(self, **updates: str) -> Transaction:
         """
         return self._append_updates(SetPropertiesUpdate(updates=updates))
 
+    def add_snapshot(self, snapshot: Snapshot) -> Transaction:
```

Comment:
I think we also need to make sure that whenever we're updating a table, we also send `AssertTableUUID` to make sure we don't update the wrong one.

Reply:
I like that, thanks!